formats/cbor/commonTest/src/kotlinx/serialization/cbor/CborReaderTest.kt

```diff
@@ -104,4 +108,343 @@ class CborReaderTest {
             )
         }
     }
+
+    @Test
+    fun testIgnoreUnknownKeysFailsWhenKotlinClassIsMissingKeys() {
```

Comment:
This test attempts to decode same CBOR encoded data as `testDecodeComplicatedObject` but attempts to decode as `Simple` rather than `TypesUmbrella`.

Validates that although `ignoreUnknownKeys` permits properties to be absent in the Kotlin class definition, the inverse (of requiring that the CBOR encoded data have all the expected fields) is still enforced.

Reply:
Summarized this PR comment as a code comment on the test function.